### PR TITLE
feat(apple): SwiftUI documentation

### PR DESCRIPTION
**Pull request by brustolin**
Documentation for SwiftUI performance monitoring.

### CODE REVIEW

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:9]: Please consider replacing this with `<Include name="alpha-note.mdx" />` or `<Include name="beta-note.mdx" />` or add an extra reusable include like the previous ones.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:28]:
```suggestion
In order to use it, you need to wrap your view with `SentryTracedView`, like this:
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:34]: `m`: It's not only the name of the transaction, it's also the name of a span if you nest them.
```suggestion
    SentryTracedView("<NAME YOUR TRANSACTION>"){
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:35]: `l`: I would add an actual view

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:40]: `m`: Please add an example for a Button or a Text.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:42]:
```suggestion
Alternatively, you can use a modifier like this:
```

Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:42]: `m`: We have to point out that this does the same as `SentryTracedView{ ... }`

File: src/wizard/apple/ios.md
Comment (philipphofmann) [inline src/wizard/apple/ios.md:129]: `m`: Instead of duplicating the info, I would add a link to the instructions above. Same applies for getting started.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:31]: `m`: I think one more example with nested views would be great.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:53]: `m`: I think we should add a screenshot of a sample transaction with nested views.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:14]: `m`: I think we should point out that SwiftUI needs an extra package. If I remember correctly, SwiftUI also includes Sentry, so we should clarify that we don't need both.

Maybe it is better to add this info to each of the pages in https://docs.sentry.io/platforms/apple/guides/ios/install/. So basically the code samples there would mention that there is also a package for SwiftUI.

```
platform :ios, '9.0'
use_frameworks! # This is important

target 'YourApp' do
  pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '7.31.5'
  # If you want SwiftUI or something like that
  pod 'SentrySwiftUI', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '7.31.5'
end
```

**Review by philipphofmann — COMMENTED** [review]
Thank you for adding docs for SwiftUI 🙏

File: src/platform-includes/getting-started-config/apple.mdx
Comment (lizokm) [inline src/platform-includes/getting-started-config/apple.mdx:119]:
```suggestion
Or by importing the sentry-cocoa repository (https://github.com/getsentry/sentry-cocoa) with SPM.
```

File: src/platform-includes/getting-started-config/apple.mdx
Comment (lizokm) [inline src/platform-includes/getting-started-config/apple.mdx:121]:
```suggestion
In order to use it, you'll need to wrap your view with `SentryTracerView`, like this:
```
Maybe I'm not understanding why this part was dropped, but if it still applies, I think it's helpful to leave it in :)

**Review by lizokm — APPROVED** [review]
Had a few small tweaks.

File: src/platform-includes/getting-started-config/apple.mdx
Comment (brustolin) [inline src/platform-includes/getting-started-config/apple.mdx:104]:
```suggestion
<Include name="beta-note.mdx" />
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:8]:
```suggestion
    <Include name="beta-note.mdx" />
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:9]:
```suggestion
<Include name="beta-note.mdx" />
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [reply src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx]: Well, that's what `Alternatively` means right?

File: src/wizard/apple/ios.md
Comment (philipphofmann) [inline src/wizard/apple/ios.md:129]:
```suggestion
If you want to find out the performance of your Views in a SwiftUI project, [try the SentrySwiftUI library](/platforms/apple/performance/instrumentation/swiftui-instrumentation).
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:11]: If you apply this suggestion you can ditch `Or by importing sentry-cocoa repository (https://github.com/getsentry/sentry-cocoa) with SPM.` below
```suggestion
You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink> or CocoaPods:
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:34]: `m`: I would add a comment as a placeholder for each code sample.
```suggestion
        List {
            // Your SwiftUI code ...
        }
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:32]: `m`: I think we could simply change this to pass in your view name, so users don't have to think about transaction names. They simply put in their view names, and Sentry does the rest. I think we should add a sentence, that the SDK will use the root view name as transaction name. We then would only need one code sample with nested views. Not two.
```suggestion
    SentryTracedView("<Name of your view>"){
```

I think we should also rename the parameter `transactionName` of `SentryTracedView`. I created an issue for that https://github.com/getsentry/sentry-cocoa/issues/2608.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:23]:
```suggestion
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:46]:
```suggestion
// Your SwiftUI code ...
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:67]:
```suggestion
                List{
                // Your SwiftUI code ...
                }
            }
            SentryTracedView("Content") {
                VStack{
                // Your SwiftUI code ...
                }
```

File: src/platform-includes/getting-started-config/apple.mdx
Comment (philipphofmann) [inline src/platform-includes/getting-started-config/apple.mdx:104]:
```suggestion
If you want to find out the performance of your Views in a SwiftUI project, [try the SentrySwiftUI library](/platforms/apple/performance/instrumentation/swiftui-instrumentation).
```

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:47]: `m`:
```suggestion
        // Your SwiftUI code ...
    }.sentryTrace("<Name of your view>")
```

**Review by philipphofmann — COMMENTED** [review]
Why did you not merge the code snippets @brustolin, as suggested in https://github.com/getsentry/sentry-docs/pull/6046#discussion_r1071015833?

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:18]: `h`: As long as we don't fix https://github.com/getsentry/sentry-cocoa/issues/2581, we should only mention SPM here, please.

Comment (brustolin) [reply src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:18]: We need to fix it before merging this documents

**Comment by brustolin** [comment]
> Why did you not merge the code snippets @brustolin, as suggested in https://github.com/getsentry/sentry-docs/pull/6046#discussion_r1071015833?

It's merged. Not sure why the conversation was not resolved.

File: src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx
Comment (brustolin) [inline src/platforms/apple/common/performance/instrumentation/swiftui-instrumentation.mdx:18]:
```suggestion
You can monitor the performance of your views in a SwiftUI project with the SentrySwiftUI library, which you can install via <PlatformLink to="/install/swift-package-manager/">SPM (Swift package manager)</PlatformLink>.
```

**Pull request merged**
